src/monitoring.rs

```diff
@@ -200,9 +203,22 @@ fn avg_metrics(metric_families: Vec<MetricFamily>) -> Option<f64> {
 
 /// Initializes performance monitoring, starting a basic HTTP server for prometheus to poll.
 pub fn init() {
-	if let Some(addr) = settings().logging.prometheus {
-		ENABLED.set(true).unwrap();
-		tokio::spawn(prometheus_server(addr));
+	if let Some(addr) = &settings().logging.prometheus {
+		match addr.to_socket_addrs() {
+			Ok(mut sock_addr) => {
+				ENABLED.set(true).unwrap();
+				tokio::spawn(prometheus_server(
+					sock_addr.next().expect("No socket address provided"),
```

Comment:
Can this ever fail? I don't think so from what I can tell so maybe something like `"String produced no socket addresses"` would be better.